bagua/torch_api/distributed.py

```diff
@@ -99,24 +99,30 @@ def with_bagua(  # pytype: disable=module-attr
         if process_group is None:
             process_group = _get_default_group()
 
-        bagua_module_name = None
-        if hasattr(self, "bagua_ddp"):
-            bagua_module_name = self.bagua_ddp.bagua_module_name
```

Comment:
In my understanding, it cannot use `.with_bagua` and `BaguaDistributedParallel` at the same time. So may not need to use separate `.bagua_module_name` to identify different modules.

Reply:
I want to make it possible for our users to tell us whether two module instances are the same or not, instead of simply being determined by the their `id`s.

In fact, this comes from a dev requirement from Lightning. A `torch.nn.Module` can be wrapped multiple times, but they share same model parameters.

Reply:
I want to make it possible for our users to tell us whether two module instances are the same or not, instead of simply being determined by the their ids.
---
Users only need to print the module.